logstash-core/src/main/java/org/logstash/execution/Filter.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.execution;
+
+import org.logstash.Event;
+
+/**
+ * A Filter is simply a mapping of {@link QueueReader} to a new {@link QueueReader}.
+ */
+public interface Filter extends AutoCloseable {
+
+    QueueReader filter(QueueReader reader);
+
+    void flush(boolean isShutdown);
```

Comment:
I don't think we want to support flushing in the future, can we remove this from V1 here?

Reply:
@andrewvc :D I'm always happen to remove `flush` why do you even ask :) Will remove it